code/modules/projectiles/guns/syringe_gun.dm

```diff
@@ -95,3 +94,249 @@
 	force = 2 //Also very weak because it's smaller
 	suppressed = 1 //Softer fire sound
 	can_unsuppress = 0 //Permanently silenced
+
+// Not quite a syringe gun, but also not completely unlike one either.
+// these names are subject to change, I just want to leave stuff in place for now.
+/obj/item/gun/syringe/rapidsyringe
+	name = "rapid syringe gun"
+	desc = "A syndicate rapid syringe gun stolen from NanoTrasen's R&D archives. Has an internal mechanism capable of storing and filling syringes from an internal reservoir.\nIt has a large flap on the side that you can dump a box or bag of syringes into, and a port for filling it with liquid."
+	icon_state = "rapidsyringegun"
+	max_syringes = 14  // full two boxes worth
+	/// Internal reagent container.
+	var/obj/item/reagent_containers/glass/beaker/bluespace/internal_beaker
+	/// Index of possible transfer amounts to select
+	var/reagents_per_syringe_selection = 1
+	/// Possible options for alt-clicking the
+	var/list/possible_transfer_amounts = list(5, 10, 15)
+
+/obj/item/gun/syringe/rapidsyringe/Initialize(mapload)
+	. = ..()
+	internal_beaker = new(src)
+
+/obj/item/gun/syringe/rapidsyringe/Destroy()
+	for(var/syringe in syringes)
+		qdel(syringe)
+
+	syringes = null
+
+	if(chambered)
+		if(chambered.BB)
+			qdel(chambered.BB)
+		qdel(chambered)
```

Comment:
shouldn't this be handled in the `/obj/item/ammo_casing`'s `Destroy()` proc?

Reply:
You'd think so, but I don't think `ammo_casing` has an overridden Destroy()`. I suppose I can make that change, but it feels like it might be a risky fix to make, given how guncode already is a little iffy